src/main/java/org/folio/circulation/domain/LoanRepresentation.java

```diff
@@ -45,13 +47,55 @@ public JsonObject extendedLoan(Loan loan) {
       extendedRepresentation.remove(LoanProperties.LOAN_POLICY);
     }
 
+    if (loan.getOverdueFinePolicy() != null) {
+      additionalOverdueFinePolicyProperties(extendedRepresentation, loan.getOverdueFinePolicy());
+    } else {
+      extendedRepresentation.remove(LoanProperties.OVERDUE_FINE_POLICY);
+    }
+
+    if (loan.getLostItemPolicy() != null) {
+      additionalLostItemPolicyProperties(extendedRepresentation, loan.getLostItemPolicy());
+    } else {
+      extendedRepresentation.remove(LoanProperties.LOST_ITEM_POLICY);
+    }
+
     additionalAccountProperties(extendedRepresentation, loan.getAccounts());
 
     extendedRepresentation.remove(LoanProperties.PATRON_GROUP_ID_AT_CHECKOUT);
 
     return extendedRepresentation;
   }
 
+  private void additionalOverdueFinePolicyProperties(JsonObject loanRepresentation,
+                                                     OverdueFinePolicy overdueFinePolicy) {
+    if (overdueFinePolicy == null) {
+      log.info("Unable to add overdue fine policy properties to loan {}," +
+        " overdueFinePolicy is null", loanRepresentation.getString("id"));
+      return;
+    }
+    JsonObject overdueFinePolicySummary = loanRepresentation.containsKey(LoanProperties.OVERDUE_FINE_POLICY)
+      ? loanRepresentation.getJsonObject(LoanProperties.OVERDUE_FINE_POLICY)
+      : new JsonObject();
+
+    overdueFinePolicySummary.put("name", overdueFinePolicy.getName());
+    loanRepresentation.put(LoanProperties.OVERDUE_FINE_POLICY, overdueFinePolicySummary);
+  }
+
+  private void additionalLostItemPolicyProperties(JsonObject loanRepresentation,
+                                                  LostItemPolicy lostItemPolicy) {
+    if (lostItemPolicy == null) {
+      log.info("Unable to add lost item policy properties to loan {}," +
+        " lostItemPolicy is null", loanRepresentation.getString("id"));
+      return;
+    }
+    JsonObject lostItemPolicySummary = loanRepresentation.containsKey(LoanProperties.LOST_ITEM_POLICY)
+      ? loanRepresentation.getJsonObject(LoanProperties.LOST_ITEM_POLICY)
+      : new JsonObject();
+
+    lostItemPolicySummary.put("name", lostItemPolicy.getName());
+
+    loanRepresentation.put(LoanProperties.LOST_ITEM_POLICY, lostItemPolicySummary);
```

Comment:
Can the duplication between this and the above method be removed? Possibly by extracting a method.

Reply:
Done